pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -487,11 +497,8 @@ func (r *ReconcileBareMetalHost) actionRegistering(prov provisioner.Provisioner,
 		info.postSaveCallbacks = append(info.postSaveCallbacks, updatedCredentials.Inc)
 	}
 
-	// Make sure we have saved the current boot mode value.
-	bootMode := info.host.BootMode()
-	if bootMode != info.host.Status.Provisioning.BootMode {
+	if updateBootModeStatus(info.host) {
```

Comment:
Unfortunately actionRegistering can get run at any time (not just right before moving to the Inspecting state), so even with this change there's no guarantee that the mode reported in the Status is the one that was actually used to provision the host. I think this would work if we did:

```
if info.Host.NeedsHardwareInspection() && !info.Host.Spec.ExternallyProvisioned && updateBootModeStatus(info.host) {
```

but it's pretty hacky :/

We'd also need to add a call to updateBootModeStatus at the transition from ExternallyProvisioned->Inspecting.

Reply:
We only need to update it before inspection and provisioning, so maybe I should move this to `actionInspecting()`? Or even inside the provisioner, since that code knows whether inspection has actually started or not.

Reply:
The state machine code could do it, too.

Reply:
I moved the logic to update the status field into the state machine, since there I could do it before transitioning into the new states for registering and provisioning.